### PR TITLE
[Modules] Updated & extended ML Workspaces API version

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Updated API version(s)
- Added 2 additional parameters 
- Added 2 additinoal allowed values for property

| Pipeline |
| - |
| [MachineLearningServices: Workspaces](https://github.com/Azure/ResourceModules/actions/workflows/ms.machinelearningservices.workspaces.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: modules/Microsoft.MachineLearningServices/workspaces/deploy.bicep [hunk 1]
```diff
@@ -131,6 +133,12 @@ param imageBuildCompute string = ''
 @sys.description('Conditional. The user assigned identity resource ID that represents the workspace identity. Required if \'userAssignedIdentities\' is not empty and may not be used if \'systemAssignedIdentity\' is enabled.')
 param primaryUserAssignedIdentity string = ''
 
+@sys.description('Optional. The service managed resource settings.')
```
Comment (eriqua) [inline modules/Microsoft.MachineLearningServices/workspaces/deploy.bicep:136]: Should these new parameters be added to the common tests?

Comment (MrMCake) [reply modules/Microsoft.MachineLearningServices/workspaces/deploy.bicep:136]: In theory yes - but I haven't had time to look into how they work. Just saw they were missing. I don't think it's too bad as we're just passing the value through without any logic - but if you think it should be required anyways I'll try to find time to figure out how I can get the PrivateLinks & CosmosDB in here.

Comment (eriqua) [reply modules/Microsoft.MachineLearningServices/workspaces/deploy.bicep:136]: Let's postpone that outside of this PR, but let's create an issue for that. I'd expect we have the same for many modules, but maybe easier to at least track the ones we notice.

Comment (eriqua) [reply modules/Microsoft.MachineLearningServices/workspaces/deploy.bicep:136]: Created #2247

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**